homeassistant/components/sensor/dominionenergy.py

```diff
@@ -0,0 +1,88 @@
+"""
+Support for Washington Dominion Energy.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dominion_energy/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+
+HOURS_TO_UPDATE = timedelta(hours=24)
+CURRENT_BILL_SELECTOR = str("#homepageContent > div:nth-child(3) >"
+                            " div:nth-child(2) > p > span")
+REQUIREMENTS = ['selenium==3.11.0']
+
+DEFAULT_NAME = "Dominion Energy"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and getting the username and password."""
+    sensor = DominionEnergySensor(config[CONF_NAME], config[CONF_USERNAME],
+                                  config[CONF_PASSWORD])
+    sensor.update()
+    if sensor.valid_credentials is True:
+        add_devices([sensor])
+    else:
+        raise PlatformNotReady
```

Comment:
This makes no sense here. If the credentials are wrong then they are wrong and the platform can't "recover". `PlatformNotReady` is useful if the service/device is not reachable.

Reply:
Ah sorry, I thought `valid_credentials` would also be `False`, if the website can't be reached ❤️